promote_packages.py

```diff
@@ -11,90 +11,152 @@
         # Packages with no link should be the ones that we mainain.
-        if package_name is not None and not has_link(client, src, package_name):
+        if (
+            package_name is not None
+            and package_name not in exclude_packages
+            and not has_link(client, src, package_name)
+        ):
             try:
                 result = get_diff(src, dst, package_name)
-                print("###################################################################", flush=True)
-                print(f"Diff for {package_name}:\n {result}", flush=True)
-                print("###################################################################", flush=True)
+                print(
+                    "###################################################################",
+                    flush=True,
+                )
+                print(f"Diff for '{package_name}' package from '{src}' to '{dst}':\n {result}", flush=True)
+                print(
+                    "###################################################################",
+                    flush=True,
+                )
                 if result != "":
-                    print(f"Copying {package_name} from {src} to {dst}\n", flush=True)
+                    print(f"Copying '{package_name}' from '{src}' to '{dst}'\n", flush=True)
                     run(["osc", "copypac", src, package_name, dst], check=True)
             except CalledProcessError:
-                print(f"Could not copypac {package_name}\n", flush=True)
-                print(format_exc(),flush=True)
+                print(f"Could not copypac '{package_name}'\n", flush=True)
+                print(format_exc(), flush=True)
                 sys.exit(1)
 
+
 def get_diff(src, dst, pkgname) -> str:
-    result = run(["osc", "rdiff", dst, pkgname, src], check=True, stdout=PIPE, stderr=PIPE)
+    result = run(
+        ["osc", "rdiff", dst, pkgname, src], check=True, stdout=PIPE, stderr=PIPE
+    )
     return result.stdout.decode("utf-8")
 
+
 def get_subprojects(client, project_name) -> list:
     prefix = project_name + ":"
     root = client.search.project("starts-with(@name,'" + prefix + "')")
     return [p.attrib["name"] for p in root.findall("project")]
 
+
 def get_project_config(client, project_name) -> str:
     project_handler = Project(client)
     return project_handler.get_config(project_name)
 
+
 def set_project_config(client, prj, config):
     project_handler = Project(client)
     project_handler.set_config(prj, config=config)
 
+
 def has_link(client, project, package) -> bool:
     pkg_handler = Package(client)
-    pkg_files = pkg_handler.get_files(project,package)
-    linkinfo = etree.fromstring(etree.tostring(pkg_files).decode('utf-8')).find("linkinfo")
+    pkg_files = pkg_handler.get_files(project, package)
+    linkinfo = etree.fromstring(etree.tostring(pkg_files).decode("utf-8")).find(
```

Comment:
Why is this "fromstring(tostring().decode("utf-8"))" needed?

Reply:
You are right. This is not really needed.